google/cloud/bigquery/_pandas_helpers.py

```diff
@@ -577,8 +577,7 @@ def _bqstorage_page_to_dataframe(column_names, dtypes, page):
 def _download_table_bqstorage_stream(
     download_state, bqstorage_client, session, stream, worker_queue, page_to_item
 ):
-    position = bigquery_storage_v1beta1.types.StreamPosition(stream=stream)
-    rowstream = bqstorage_client.read_rows(position).rows(session)
+    rowstream = bqstorage_client.read_rows(stream.name).rows(session)
```

Comment:
Can we detect when a v1beta1 client is passed in and still use `StreamPosition`? Otherwise, this is a breaking change for `pandas-gbq` and customer code that used the BQ Storage API before this change.

Reply:
We can check the client type, yes, and construct the read session and its parameters differently. It should only be necessary to have this check at a few limited places, thus the extra noise should not be too bad, especially if we prevent a severe breaking change.

Reply:
Supporting a transition period where we support either storage client makes sense.  I'd love to emit deprecation warnings for the v1beta1 client case so people can identify they should migrate sooner rather than later.  We should pick a reasonable drop dead date for this as well so we're not carrying it indefinitely if we can avoid it.

A quick scan of api core blunted my hopes of trying to do something clever like using a v1beta1 client to build a v1 client, it looks like there's too much we could get wrong with transports and such.

Reply:
Done, and seems to work (except for the deprecation warning that I still need to add).